InertiaCore/Utils/Vite.cs

```diff
@@ -0,0 +1,256 @@
+using System.Text.Encodings.Web;
+using System.Text.Json;
+using Microsoft.AspNetCore.Html;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using System.Text.RegularExpressions;
+
+namespace InertiaCore.Utils;
+
+
+class ViteBuilder
+{
+    // The path to the "hot" file.
+    public string hotFile = "hot";
+
+    // The path to the build directory.
+    public string? buildDirectory = "build";
+
+    // The name of the manifest file.
+    public string manifestFilename = "manifest.json";
+
+    // The path to the public directory.
+    public string publicDirectory = "wwwroot";
+
+    public static ViteBuilder? instance = null;
+    public static ViteBuilder Instance
+    {
+        get
+        {
+            if (instance == null)
+            {
+                instance = new ViteBuilder();
+            }
```

Comment:
It can be shortened to just:
```suggestion
            instance ??= new ViteBuilder();
```